assets/src/components/form/inputs/base/utils.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @function
+ * @param {Object} attributes
+ * @return {Object} attributes
+ */
+export const cleanUpInputAttributes = ( attributes = {} ) => {
+	delete attributes.children;
+	delete attributes.colSize;
+	delete attributes.helpText;
+	delete attributes.helpTextID;
+	delete attributes.label;
+	delete attributes.render;
+	delete attributes.validations;
+	return attributes;
+};
+
+/**
+ * @function
+ * @param {Array} validators
+ * @return {Function} combined validation function
+ */
+export const composeValidators = ( ...validators ) => value => {
+	return validators.reduce(
+		( error, validator ) => error || validator( value ),
+		undefined
+	);
+};
+
+/**
+ * @function
+ * @param {Object} attributes
+ * @param {Array} validators
+ * @return {Object} attributes
+ */
+export const mapValidatorsToAttributes = (
```

Comment:
ya the intent was to include other attributes. I had tried to do so for the `min` and `max` attributes but because of the way the `maxNumber` and `minNumber` validators are structured as higher-order-functions, the name property was blank.

ex:

```js
export const minNumber = min => value => {
	min = parseInt( min, 10 );
	return parseInt( value, 10 ) < min ?
		sprintf(
			_n(
				'Number should be greater than %d',
				'Number should be greater than %d',
				min,
				'event_espresso'
			),
			min
		) : null;
};
```

The outer function can be detected by name because that is set (minNumber), but that function returns an unnamed inner function which is actually what gets passed to `React-Final-Form` as the validation function to run. The outer function merely sets the accepted minimum value for the number input when the form is first constructed.

I tried to split this up so that the inner function was also a named function but that didn't work.

Anyways, I left this as is for the time being in case there are other validators we want to detect and convert into attributes.

Reply:
hmm okay, I'm still not a fan of the name though mostly because it appears to be mutating the attribute values (is that correct)?

Reply:
how about `addValidatorsToAttributes()` ?

Reply:
ya that's a bit better.